### ISSUE TITLE
moreh.layer_norm mean and rstd initial shape

### ISSUE THREAD

**Issue by kevinwuTT** [tenstorrent/tt-metal#22110]
### Is your feature request related to a problem? Please describe

A similar function to `moreh.layer_norm` is `torch.ops.aten.native_layer_norm.default`, which also returns three outputs including mean and rstd. However, the normalized dimensions of the mean and rstd outputs are kept. For example, in this code, the output shapes are [2, 32, 512], [2, 32, 1], [2, 32, 1]. 

```
import torch

torch.manual_seed(0)
input = torch.rand((2, 32, 512), dtype=torch.bfloat16)
normalized_dims = 1
normalized_shape = input.shape[-normalized_dims:]
weight = torch.rand((normalized_shape), dtype=torch.bfloat16)
bias = torch.rand((normalized_shape), dtype=torch.bfloat16)
eps = 1e-05

out, mean, rstd = torch.ops.aten.native_layer_norm.default(input, normalized_shape, weight, bias, eps)
print(out.shape)
print(mean.shape)
print(rstd.shape)
```

For `moreh.layer_norm`, I tried to preallocate the mean and rstd tensors with zeros and with the same shape as the torch version: [2, 31, 1]. However, most of the values in `mean` and `rstd` remain zeros. Only a few indices changed. When I preallocate the tensors without the normalized dimension, [2, 31], the `mean` and `rstd` tensors look correct. 

```
import ttnn
import torch

torch.manual_seed(0)
input = torch.rand((2, 32, 512), dtype=torch.bfloat16)
normalized_dims = 1
weight = torch.rand((2, 32), dtype=torch.bfloat16)
bias = torch.rand((2, 32), dtype=torch.bfloat16)
eps = 1e-05

with ttnn.manage_device(device_id=0) as device:
    ttnn_input = ttnn.from_torch(input, device = device, layout = ttnn.TILE_LAYOUT, dtype = ttnn.bfloat16)
    ttnn_weight = ttnn.from_torch(weight, device = device, layout = ttnn.TILE_LAYOUT, dtype = ttnn.bfloat16)
    ttnn_bias = ttnn.from_torch(bias, device = device, layout = ttnn.TILE_LAYOUT, dtype = ttnn.bfloat16)
    
    ttnn_out = ttnn.empty(ttnn_input.shape, device=device, layout = ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16)
    
    mean_rstd_shape = (2, 32, 1)
    ttnn_mean = ttnn.zeros(mean_rstd_shape, device=device, layout = ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16)
    ttnn_rstd = ttnn.zeros(mean_rstd_shape, device=device, layout = ttnn.TILE_LAYOUT, dtype=ttnn.bfloat16)
    
    ttnn_out, ttnn_mean, ttnn_rstd = ttnn.operations.moreh.layer_norm(
        ttnn_input, normalized_dims, eps, ttnn_weight, ttnn_bias,
        output=ttnn_out,
        mean=ttnn_mean,
        rstd=ttnn_rstd
    )
   
    ttnn_mean = ttnn.to_torch(ttnn_mean, dtype = torch.bfloat16)
    ttnn_rstd = ttnn.to_torch(ttnn_rstd, dtype = torch.bfloat16)
    
    print(ttnn_mean)
    print(ttnn_rstd)
```

### Describe the solution you'd like

Is there confirmation that this is intended behavior or possibly a bug? If the shapes for `mean` and `rstd` are incompatible, there perhaps some error messages would be useful.

### Describe alternatives you've considered

If this is intentional, then I could add reshapes after calling `moreh.layer_norm`. But would like to have some confirmation. 

### Additional context

This is related to a compiler for transforming `torch.aten` ops to `ttnn` ops. `moreh.layer_norm` is the closest to `torch.ops.aten.native_layer_norm.default` which returns 3 outputs. However, the Moreh version differs slightly when it comes to the output shapes for `mean` and `rstd`.
